### PR TITLE
examples(FileSource): simplifies examples on FileSource usage

### PR DESCRIPTION
**Pull request by mgermerie**
## Description

Simplifies examples on `FileSource` usage :
- add examples on different ways of instantiating `FileSource`
- simplify `FileSource` instantiation in other examples.

### CODE REVIEW

File: examples/source_file_from_fetched_data.html [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+        config.source = new itowns.WMTSSource(config.source);
+        var layer = new itowns.ColorLayer('Ortho', config);
+        view.addLayer(layer).then(debugMenu.addLayerGUI.bind(debugMenu));
+    });
+
+
+
+    // ---------- INSTANTIATE FileSource FROM FETCHED DATA : ----------
+
+    // First, fetch the data. Should you implement a custom fetcher, it would need to be called here.
```
Comment (gchoqueux) [inline examples/source_file_from_fetched_data.html]: add comment `handle fetching step`

File: examples/source_file_from_parsed_data.html [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                crs: 'EPSG:4326',
+            },
+            // Information on the FeatureCollection that should be created from the fetched data. Here, we pass a
+            // FeatureBuildingOptions (http://www.itowns-project.org/itowns/docs/#api/Base/FeatureBuildingOptions).
+            out: {
+                crs: view.tileLayer.extent.crs,
+                buildExtent: true,
+                mergeFeatures: true,
+                structure: '2d',
+            },
```
Comment (gchoqueux) [inline examples/source_file_from_parsed_data.html]: could you remove or add default values with the structure parameter ?
buildExtent: true,
mergeFeatures: true,

could you add more comment to explain other parameters

File: examples/source_file_kml_raster.html [hunk 3]
```diff
@@ -79,7 +78,7 @@
 
             var kmlLayer = new itowns.ColorLayer('Kml', {
                 name: 'kml',
-                transparent: true,
+                transparent: true,  // TODO : could be removed
```
Comment (gchoqueux) [inline examples/source_file_kml_raster.html]: could you try?

Comment (gchoqueux) [reply examples/source_file_kml_raster.html]: this parameter is used?

Comment (gchoqueux) [reply examples/source_file_kml_raster.html]: and how it's used?

**Review by gchoqueux — COMMENTED** [review]
Could you rename 

- `Source / Stream` to `WFS source`
- `Source / File - Instantiate FileSource` separate to
    - `Customize File source`
      - handle fetching step 
      - handle parsing step
    - `Specific option source` 
       - options `fetcher` and `parser`
- `Source / File - Different file formats` to `File Source`


`GeoJSON to raster` with tool tip -> copy in plugin

Remove all `INSTANCIATE` comment and replace by `define your source` (by example)

File: examples/source_file_from_fetched_data.html (hunk 1 above)
Comment (mgermerie) [reply examples/source_file_from_fetched_data.html]: Done.

File: examples/source_file_from_parsed_data.html (hunk 2 above)
Comment (mgermerie) [reply examples/source_file_from_parsed_data.html]: The default values were added in PR #1752.

**Comment by mgermerie** [comment]
> Could you rename
> 
>     * `Source / Stream` to `WFS source`
> 
>     * `Source / File - Instantiate FileSource` separate to
>       
>       * `Customize File source`
>         
>         * handle fetching step
>         * handle parsing step
>       * `Specific option source`
>         
>         * options `fetcher` and `parser`
> 
>     * `Source / File - Different file formats` to `File Source`
> 
> 
> `GeoJSON to raster` with tool tip -> copy in plugin
> 
> Remove all `INSTANCIATE` comment and replace by `define your source` (by example)

I changed the examples accordingly, except for the `GeoJSON to raster` in plugins part that we choose to abort.

File: examples/config.json [hunk 4]
```diff
@@ -28,19 +28,29 @@
...
+    },
+
+    "FileSource": {
         "source_file_geojson_raster": "GeoJSON to raster",
+        "source_file_geojson_3d": "GeoJSON to 3D objects",
+        "source_file_kml_raster": "KML to raster",
         "source_file_gpx_raster": "GPX to raster",
-        "source_file_kml_raster": "KML to raster"
+        "source_file_gpx_3d": "GPX to 3D objects",
+        "source_file_shapefile": "Shapefile to raster"
```
Comment (gchoqueux) [inline examples/config.json]: source_file_shapefile move to `Customize FileSource` and rename to `shape file case`

 "Customize FileSource" move after  "FileSource"

**Review by gchoqueux — APPROVED** [review]

File: examples/config.json (hunk 4 above)
Comment (mgermerie) [reply examples/config.json]: I changed the config accordingly.

**Pull request merged**